jupyter_bokeh/widgets.py

```diff
@@ -126,13 +127,13 @@ def _sync_model(self, _, content, _buffers):
             submodel = self._model.select_one({"id": content["id"]})
             descriptor = submodel.lookup(content['attr'])
             try:
-                descriptor._real_set(submodel, old, new, hint=hint, setter=self)
+                descriptor._set(submodel, old, new, hint=hint, setter=self)
             except Exception:
                 return
             for cb in submodel._callbacks.get(attr, []):
                 cb(attr, old, new)
         elif kind == 'MessageSent':
-            self._document.apply_json_event(content["msg_data"])
+            self._document.callbacks.trigger_json_event(content["msg_data"])
```

Comment:
This is frustrating, this method was public.

Reply:
I thought I telegraphed these changes pretty loudly, these PRs happened over the course of several weeks. Also the stated position in the past has been "Don't worry about breaking changes, we pin Bokeh minor versions". Yes this was "public" but this code here is probably literally the only consumer of that API anywhere in the world. The real issue is that there are no downstream tests between Bokeh and `jupyter_bokeh`. I am sorry you are frustrated. I am doing the best I can.

Reply:
I'm sorry I wasn't more careful in my phrasing and I know you are very diligent with all your changes. You're right that downstream tests are missing here and also that this was telegraphed well ahead of time. I'm also trying to keep up with 10 different projects and doing the best I can 🙂

Reply:
Indeed the real breakage here was in private API so this Widget would have been broken regardless.

Reply:
I am going to try and prioritize CI, automation, and testing for this repo over the next few weeks.